src/components/PopupWithForm.js

```diff
@@ -5,6 +5,19 @@ class PopupWithForm extends Popup {
     super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
     this._popupForm = this._popupElement.querySelector(".modal__form");
+    console.log(this._popupElement); // Check if this logs the correct element
+    if (!this._popupElement) {
+      throw new Error(`Element not found for selector: ${selector}`);
+    }
+  }
+
```

Comment:
**COULD BE IMPROVED**

You can make a special method for getting a form element from `PopupWithForm` not to **search** it in `index.js`



```js

  getForm() {
    return this._popupForm
  }


```